From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Euphyllia <bierqueeuphyllia@gmail.com>
Date: Sat, 4 May 2024 21:47:03 +0200
Subject: [PATCH] Bloom Config Files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index d4c5b06fe4be177442ef1abaf1b223c6c2845930..349f245a70624a7a6d84edc4b6fd821bbf9ae2dc 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2257,6 +2257,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // Bloom start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getBloomConfig()  {
+            throw new UnsupportedOperationException("Not supported yet");
+        }
+        // Bloom end
+
         /**
          * Sends the component to the player
          *
