From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Sun, 5 May 2024 03:47:18 +0200
Subject: [PATCH] PlayerTeleportEvent and EntityTeleportEvent


diff --git a/src/main/java/fr/euphyllia/bloom/configuration/GlobalConfiguration.java b/src/main/java/fr/euphyllia/bloom/configuration/GlobalConfiguration.java
index eb618442c99114c3b81c2522242944dad58654f8..61465b5a34528fec951ad030d8e6004f6cf40cd8 100644
--- a/src/main/java/fr/euphyllia/bloom/configuration/GlobalConfiguration.java
+++ b/src/main/java/fr/euphyllia/bloom/configuration/GlobalConfiguration.java
@@ -116,4 +116,9 @@ public class GlobalConfiguration {
         }
         return builder.build();
     }
+
+    public static boolean ENABLE_ENTITY_TELEPORT_EVENT = false;
+    private static void teleportCalled() {
+        ENABLE_ENTITY_TELEPORT_EVENT = getBoolean("event.entity.teleport_event", ENABLE_ENTITY_TELEPORT_EVENT);
+    }
 }
\ No newline at end of file
diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 7a102b4e90fbc09b8653e5b566299efe24a04cf6..557774a02ee6b863085287c01077814480569b8f 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -4073,6 +4073,25 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
         }
 
         // TODO any events that can modify go HERE
+        // Bloom start
+        if (fr.euphyllia.bloom.configuration.GlobalConfiguration.ENABLE_ENTITY_TELEPORT_EVENT) {
+            org.bukkit.entity.Entity bukkitEntity = this.getBukkitEntity();
+            Location to = new Location(destination.getWorld(), blockPos.getX(), blockPos.getY(), blockPos.getZ(), yaw != null ? yaw : 0.0f, pitch != null ? pitch : 0.0f);
+            if (bukkitEntity instanceof org.bukkit.entity.Player player) {
+                PlayerTeleportEvent playerTeleportEvent = new PlayerTeleportEvent(player, bukkitEntity.getLocation(), to);
+                destination.getCraftServer().getPluginManager().callEvent(playerTeleportEvent);
+                if (playerTeleportEvent.isCancelled()) {
+                    return false;
+                }
+            } else {
+                org.bukkit.event.entity.EntityTeleportEvent entityTeleportEvent = new org.bukkit.event.entity.EntityTeleportEvent(bukkitEntity, bukkitEntity.getLocation(), to);
+                destination.getCraftServer().getPluginManager().callEvent(entityTeleportEvent);
+                if (entityTeleportEvent.isCancelled()) {
+                    return false;
+                }
+            }
+        }
+        // Bloom end
 
         // check for same region
         if (destination == this.level()) {
